cypress/integration/console/shared/api-keys/edit.spec.js

```diff
@@ -0,0 +1,199 @@
+
+      cy.findByTestId('modal-window')
+        .should('be.visible')
+        .within(() => {
+          cy.findByText('Delete key', { selector: 'h1' }).should('be.visible')
+          cy.findByRole('button', { name: /Delete key/ }).click()
+        })
+
+      cy.findByTestId('error-notification').should('not.exist')
+
+      cy.location('pathname').should(
+        'eq',
+        `${Cypress.config('consoleRootPath')}/applications/${applicationId}/api-keys`,
+      )
+
+      cy.findByRole('cell', { name: apiKeyName }).should('not.exist')
+    })
+  })
+
+  describe('Gateway', () => {
+    const gatewayId = 'api-keys-test-gateway'
+    const gateway = { ids: { gateway_id: gatewayId } }
+
+    before(() => {
+      cy.loginConsole({ user_id: userId, password: user.password })
+      cy.createGateway(gateway, userId)
+      cy.createApiKey('gateways', gatewayId, apiKey)
+      cy.clearLocalStorage()
+      cy.clearCookies()
+    })
+
+    beforeEach(() => {
+      cy.loginConsole({ user_id: user.ids.user_id, password: user.password })
+    })
+
+    it('succeeds editing api key', () => {
+      cy.visit(`${Cypress.config('consoleRootPath')}/gateways/${gatewayId}/api-keys`)
+
+      cy.findByRole('cell', { name: apiKeyName }).click()
+      cy.findByRole('button', { name: 'Save changes' }).click()
+
+      cy.findByTestId('toast-notification')
+        .should('be.visible')
+        .findByText(`API key updated`)
+        .should('be.visible')
+    })
+
+    it('succeeds deleting api key', () => {
+      cy.visit(`${Cypress.config('consoleRootPath')}/gateways/${gatewayId}/api-keys`)
+
+      cy.findByRole('cell', { name: apiKeyName }).click()
+      cy.findByRole('button', { name: /Delete key/ }).click()
+
+      cy.findByTestId('modal-window')
+        .should('be.visible')
+        .within(() => {
+          cy.findByText('Delete key', { selector: 'h1' }).should('be.visible')
+          cy.findByRole('button', { name: /Delete key/ }).click()
+        })
+
+      cy.findByTestId('error-notification').should('not.exist')
+
+      cy.location('pathname').should(
+        'eq',
+        `${Cypress.config('consoleRootPath')}/gateways/${gatewayId}/api-keys`,
+      )
+
+      cy.findByRole('cell', { name: apiKeyName }).should('not.exist')
+    })
+  })
+
+  describe('Organization', () => {
+    const organizationId = 'api-keys-test-org'
+    const organization = {
+      ids: { organization_id: organizationId },
+    }
+
+    before(() => {
+      cy.loginConsole({ user_id: userId, password: user.password })
+      cy.createOrganization(organization, userId)
+      cy.createApiKey('organizations', organizationId, apiKey)
+      cy.clearLocalStorage()
+      cy.clearCookies()
+    })
+
+    beforeEach(() => {
+      cy.loginConsole({ user_id: user.ids.user_id, password: user.password })
+    })
+
+    it('succeeds editing api key', () => {
+      cy.visit(`${Cypress.config('consoleRootPath')}/organizations/${organizationId}/api-keys`)
```

Comment:
Navigate to the edit page directly. `/console/applications/${applicationId}/api-keys/${apiKeyId}`.